### ISSUE TITLE
PBMC error when trying to view RNA Assay

### ISSUE THREAD

**Issue by matthewbradley22** [satijalab/seurat#8022]
If I read in the PBMC data from the Seurat vignette, create a Seurat object, and try to extract contents of the RNA assay using the "@" symbol, I get the following error:

"Error in names(x) <- paste("V", seq_along(x), sep = "") :  'names' attribute [1] must be the same length as the vector [0]".

```r
pbmc.data <- Read10X("Path to downloaded data from the Seurat vignette")
pbmc <- CreateSeuratObject(pbmc.data)
pbmc@assays$RNA@
```

The moment I type the second "@" symbol in the above code, I get the aforementioned error.

**Comment by samuel-marsh** [comment]
Hi,

Not member of the dev team but hopefully can be helpful.  Can you provide the output of `sessionInfo()`?  Also are you running this in RStudio or R terminal?  Finally, are you saying that without entering a command but just typing the second @ you are getting an error?

I cannot replicate your error on my end so any details would be helpful.

Best,
Sam

**Comment by matthewbradley22** [comment]
Hello,

thanks for your response!

Session info returns:

`R version 4.3.0 (2023-04-21)
Platform: x86_64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.3.1

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] SingleCellExperiment_1.22.0 speckle_1.0.0               org.Mm.eg.db_3.17.0        
 [4] topGO_2.52.0                SparseM_1.81                GO.db_3.17.0               
 [7] AnnotationDbi_1.62.2        graph_1.78.0                celldex_1.10.1             
[10] SingleR_2.2.0               SummarizedExperiment_1.30.2 Biobase_2.60.0             
[13] GenomicRanges_1.52.1        GenomeInfoDb_1.36.4         IRanges_2.34.1             
[16] S4Vectors_0.38.2            BiocGenerics_0.46.0         MatrixGenerics_1.12.3      
[19] matrixStats_1.1.0           Seurat_5.0.0                SeuratObject_5.0.0         
[22] sp_2.1-1                   

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.21              splines_4.3.0                 later_1.3.1                  
  [4] bitops_1.0-7                  filelock_1.0.2                R.oo_1.25.0                  
  [7] tibble_3.2.1                  polyclip_1.10-6               fastDummies_1.7.3            
 [10] lifecycle_1.0.4               edgeR_3.42.4                  globals_0.16.2               
 [13] lattice_0.22-5                MASS_7.3-60                   magrittr_2.0.3               
 [16] limma_3.56.2                  plotly_4.10.3                 yaml_2.3.7                   
 [19] httpuv_1.6.12                 sctransform_0.4.1             spam_2.10-0                  
 [22] spatstat.sparse_3.0-3         reticulate_1.34.0             cowplot_1.1.1                
 [25] pbapply_1.7-2                 DBI_1.1.3                     RColorBrewer_1.1-3           
 [28] abind_1.4-5                   zlibbioc_1.46.0               Rtsne_0.16                   
 [31] R.utils_2.12.2                purrr_1.0.2                   RCurl_1.98-1.13              
 [34] rappdirs_0.3.3                GenomeInfoDbData_1.2.10       ggrepel_0.9.4                
 [37] irlba_2.3.5.1                 listenv_0.9.0                 spatstat.utils_3.0-4         
 [40] goftest_1.2-3                 RSpectra_0.16-1               spatstat.random_3.2-1        
 [43] fitdistrplus_1.1-11           parallelly_1.36.0             DelayedMatrixStats_1.22.6    
 [46] leiden_0.4.3                  codetools_0.2-19              DelayedArray_0.26.7          
 [49] tidyselect_1.2.0              ScaledMatrix_1.8.1            BiocFileCache_2.8.0          
 [52] spatstat.explore_3.2-5        jsonlite_1.8.7                ellipsis_0.3.2               
 [55] progressr_0.14.0              ggridges_0.5.4                survival_3.5-7               
 [58] tools_4.3.0                   ica_1.0-3                     Rcpp_1.0.11                  
 [61] glue_1.6.2                    gridExtra_2.3                 dplyr_1.1.3                  
 [64] BiocManager_1.30.22           fastmap_1.1.1                 fansi_1.0.5                  
 [67] digest_0.6.33                 rsvd_1.0.5                    R6_2.5.1                     
 [70] mime_0.12                     colorspace_2.1-0              scattermore_1.2              
 [73] tensor_1.5                    spatstat.data_3.0-3           RSQLite_2.3.3                
 [76] R.methodsS3_1.8.2             utf8_1.2.4                    tidyr_1.3.0                  
 [79] generics_0.1.3                data.table_1.14.8             httr_1.4.7                   
 [82] htmlwidgets_1.6.2             S4Arrays_1.0.6                uwot_0.1.16                  
 [85] pkgconfig_2.0.3               gtable_0.3.4                  blob_1.2.4                   
 [88] lmtest_0.9-40                 XVector_0.40.0                htmltools_0.5.7              
 [91] dotCall64_1.1-0               scales_1.2.1                  png_0.1-8                    
 [94] rstudioapi_0.15.0             reshape2_1.4.4                nlme_3.1-163                 
 [97] curl_5.1.0                    zoo_1.8-12                    cachem_1.0.8                 
[100] stringr_1.5.0                 BiocVersion_3.17.1            KernSmooth_2.23-22           
[103] parallel_4.3.0                miniUI_0.1.1.1                pillar_1.9.0                 
[106] grid_4.3.0                    vctrs_0.6.4                   RANN_2.6.1                   
[109] promises_1.2.1                BiocSingular_1.16.0           dbplyr_2.4.0                 
[112] beachmat_2.16.0               xtable_1.8-4                  cluster_2.1.4                
[115] locfit_1.5-9.8                cli_3.6.1                     compiler_4.3.0               
[118] rlang_1.1.2                   crayon_1.5.2                  future.apply_1.11.0          
[121] plyr_1.8.9                    stringi_1.8.1                 viridisLite_0.4.2            
[124] deldir_1.0-9                  BiocParallel_1.34.2           munsell_0.5.0                
[127] Biostrings_2.68.1             lazyeval_0.2.2                spatstat.geom_3.2-7          
[130] Matrix_1.6-1.1                ExperimentHub_2.8.1           RcppHNSW_0.5.0               
[133] patchwork_1.1.3               sparseMatrixStats_1.12.2      bit64_4.0.5                  
[136] future_1.33.0                 ggplot2_3.4.4                 KEGGREST_1.40.1              
[139] shiny_1.7.5.1                 interactiveDisplayBase_1.38.0 AnnotationHub_3.8.0          
[142] ROCR_1.0-11                   igraph_1.5.1                  memoise_2.0.1                
[145] bit_4.0.5      `

I am working in RStudio. And yes, I get an error as soon as I type the "@" symbol, which I've never seen in my several years of using R. 

I am running code in a script that worked as recently as last wednesday, so I am guessing the error is due to some external change (some package update etc..), but I'm not sure.

**Comment by samuel-marsh** [comment]
Ya I've never seen an error occur just from typing out command and I can't replicate on my end.  Have you tried both removing and reinstalling RStudio and removing Seurat/SeuratObject and reinstalling?

Best,
Sam

**Comment by matthewbradley22** [comment]
Hello,

I fixed the issue by reinstalling Seurat v 4.4.0, and restarting R-Studio!

 I'm going to link to a vimeo video of the error happening on my screen because it's so odd that I happened when I typed the "@" symbol, and maybe it will be useful for a dev (or anyone else) to see. Thanks for your help!

 [Video](https://vimeo.com/884561209?share=copy) . I show that I can view the Seurat object, and the assays within it, but as soon as I type the "@" to extract any content, I receive the error.

**Comment by mhkowalski** [comment]
That's super interesting, thanks for reporting! We'll keep an eye out for this. I'm glad that reinstalling fixed the issue for you. Just as a general comment, we recommend that you use [GetAssayData](https://mojaveazure.github.io/seurat-object/reference/AssayData.html) (or ```LayerData``` in Seurat5) to access the contents of a particular assay.
